glean-core/src/upload/directory.rs

```diff
@@ -45,87 +38,154 @@ fn get_file_name_as_str(path: &Path) -> Option<&str> {
-            return Some(PingRequest::new(uuid, &url, parsed_body));
-        } else {
-            log::warn!("Can't parse ping contents as JSON.");
+impl PingDirectoryManager {
+    /// Creates a new directory manager.
+    ///
+    /// # Arguments
+    ///
+    /// * `data_path` - Path to the pending pings directory.
+    ///
+    /// # Panics
+    ///
+    /// Will panic if unable to parse the data_path.
+    pub fn new(data_path: &str) -> Self {
+        Self {
+            data_path: PathBuf::from_str(data_path).expect("data_path must be a valid path."),
         }
-    } else {
-        log::warn!("Ping file is not formatted as expected.");
     }
-    delete_file(path);
-    None
-}
 
-/// Process the pings directory and return a vector of `PingRequest`s
-/// corresponding to each valid ping file in the directory.
-/// This vector will be ordered by file `modified_date`.
-///
-/// Any files that don't match the UUID regex will be deleted
-/// to prevent files from polluting the pings directory.
-///
-/// Files that are not correctly formatted will also be deleted.
-pub fn process_pings_dir(data_path: &Path) -> Result<Vec<PingRequest>> {
-    let pings_dir = get_pings_dir(data_path);
-    let mut pending_pings: Vec<_> = pings_dir
-        .read_dir()?
-        .filter_map(|entry| entry.ok())
-        .filter_map(|entry| {
-            let path = entry.path();
-            if let Some(file_name) = get_file_name_as_str(&path) {
-                // Delete file if it doesn't match the pattern.
-                if Uuid::parse_str(file_name).is_err() {
-                    log::warn!("Pattern mismatch {}", path.display());
-                    delete_file(&path);
-                    return None;
-                }
-                // In case we can't process the file we just ignore it.
-                if let Some(request) = process_ping_file(file_name, &path) {
-                    // Get the modified date of the file, which will later be used
-                    // for sorting the resulting vector.
-                    let modified_date = fs::metadata(&path).and_then(|data| data.modified());
-                    return Some((modified_date, request));
-                }
-            };
-            None
-        })
-        .collect();
-
-    // Sort by `modified_date`.
-    pending_pings.sort_by(|(a, _), (b, _)| {
-        // We might not be able to get the modified date for a given file,
-        // in which case we just put it at the end.
-        if let (Ok(a), Ok(b)) = (a, b) {
-            a.partial_cmp(b).unwrap()
+    /// Get the pending pings directory path.
+    pub fn get_dir(&self) -> PathBuf {
+        self.data_path.join("pending_pings")
+    }
+
+    /// Attempts to delete a ping file.
+    ///
+    /// ## Arguments
+    ///
+    /// * `uuid` - The UUID of the ping file to be processed
+    ///
+    /// ## Panics
+    ///
+    /// Won't panic if unable to delete the file.
+    pub fn delete_file(&self, uuid: &str) {
```

Comment:
Does this need to be public?

Reply:
Yes, it is used in the `process_ping_upload_response` function.